### PR TITLE
Feat: authorization

### PR DESCRIPTION
**Pull request by Imagine96**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+> **_NOTE:_** Add SECRET_KEY="sklajdgb34tbksdjavfSECREt" to env in development for jwt functions
```
Comment (arol) [inline README.md:1]: It doesn't matter what secret content is it right?

A good practice is to create a `.env.default` file that contains all the ENV vars filled in, for the ones that are meaningful you put a template, for the ones that aren't like this one you add the key already.

File: src/lib/middleware/withAuth.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { decode, encode } from "@/services/auth/jwt";
+import type { NextApiRequest, NextApiResponse, NextApiHandler } from "next";
+import { IApiResponse, IUser } from "../types";
+
+export const enforceBearerToken = (handler: NextApiHandler) => {
+    return async (req: NextApiRequest, res: NextApiResponse<IApiResponse<unknown>>) => {
+        const { userId } = req.query
```
Comment (arol) [inline src/lib/middleware/withAuth.ts:7]: Why are you getting the `userId` from the querystring?

**Review by arol — COMMENTED** [review]
Just want to clarify first why you're enforcing to pass the userId in the `req.query`.
All requests that need to be authenticated should have the own user id in the url? What if I'm trying to mark someone else's ticket in the favorites, or an event to the watchlist. 

It seems like you're putting authentication and authorization together.

File: README.md (hunk 1 above)
Comment (Imagine96) [reply README.md:1]: and that default i can add to the commit so everybody pulling will have the dev default config right?

Comment (arol) [reply README.md:1]: Yes, the idea is to share the template. Any real secrets should not be commited.

**Review by arol — APPROVED** [review]

**Pull request merged**